src/exml_query.erl

```diff
@@ -21,16 +21,24 @@
 -export([cdata/1]).
 
 -type element_with_ns() :: {element_with_ns, binary()}.
--type element_with_name_and_ns () :: {element_with_ns, binary(), binary()}.
--type element_with_attr_of_value () :: {element_with_attr, binary(), binary()}.
+-type element_with_name_and_ns() :: {element_with_ns, binary(), binary()}.
+-type element_with_attr_of_value() :: {element_with_attr, binary(), binary()}.
+-type path() :: [cdata |
+                 {element, binary()} |
+                 {attr, binary()} |
+                 element_with_ns() |
+                 element_with_name_and_ns() |
+                 element_with_attr_of_value()].
 
--type path() :: [cdata | %% selects cdata from the element
-                 {element, binary()} | % selects subelement with given name
-                 {attr, binary()} | % selects attr of given name
-                 element_with_ns() | % selects subelement with given namespace
-                 element_with_name_and_ns() | % selects subelement with given name and namespace
-                 element_with_attr_of_value() % selects subelement with given attribute and value
-                ].
+                                                %% selects cdata from the element
```

Comment:
The comments are scattered around in a mess. Expected: comments left intact on their respective lines.

Reply:
Yeah, this is a known issue. Since it's very hard to solve, we decided that (unless someone writes a PR solving it) for now the formatter can't deal with interleaved comments in type specs. The workaround is to put them all together above the type definition, like a single long comment block.
We know it's not ideal, but don't keep your hopes up for a solution. 🤷‍♂

Reply:
In any case, if we rewrite this comment as follows, then it will also be picked up by EDoc and [exported to the Docs chunk](https://github.com/erlef/eef-documentation-wg/issues/4):

```erlang
-type path() :: [cdata |
                 {element, binary()} |
                 {attr, binary()} |
                 element_with_ns() |
                 element_with_name_and_ns() |
                 element_with_attr_of_value()
                ].
%% `path()' describes a relative position of a node in an XML document. The allowed path components are: 
%% - `cdata' - selects cdata from the element.
%% - `{element, binary()}' - selects a subelement with a given name.
%% - `{attr, binary()}' - selects an attribute of a given name.
%% - `element_with_*' - select a subelement with given parameters.
```

The tools are there to help us, but sometimes we have to help the tools help us ;)